src/main/java/teammates/ui/template/FeedbackSessionPublishButton.java

```diff
@@ -0,0 +1,83 @@
+package teammates.ui.template;
+
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.Sanitizer;
+import teammates.ui.controller.PageData;
+
+public class FeedbackSessionPublishButton {
+
+    private boolean hasUnpublish;
+    private boolean hasPublish;
+
+    private String tooltipText;
+
+    private String unpublishLink;
+    private String publishLink;
+
+    private boolean allowedToUnpublish;
+    private boolean allowedToPublish;
+
+    private String toggleUnpublishSessionParams;
+    private String togglePublishSessionParams;
+    
+    public FeedbackSessionPublishButton(PageData data, FeedbackSessionAttributes session,
+                                        boolean isHome, InstructorAttributes instructor) {
+        String courseId = session.courseId;
+        String feedbackSessionName = session.feedbackSessionName;
+
+        this.hasUnpublish = !session.isWaitingToOpen() && session.isPublished();
+        this.hasPublish = !session.isWaitingToOpen() && !session.isPublished();
```

Comment:
I've read through this part of the code before and always found this confusing because it is actually possible for the button to have hasUnpublish and hasPublish to both be false. It is not very apparent what happens when both are false until we realize from reading a lot deeper that if both are false, we get a disabled publish button.

I'm thinking if there's a better way to flatten out the logic so that it's easier to read this part of the code.